### ISSUE TITLE
Question on reproducing the evaluation/demo performance from pretrained models

### ISSUE THREAD

**Issue by hetong007** [mbzuai-oryx/groundingLMM#20]
Hello, first of all thanks for the great work!

I've been trying to reproduce the evaluation and demo, however I don't find them producing the same quality results as produced in official materials.

**Environment**
- CUDA 12.1, PyTorch 2.1.2+cu121
- A100 with 40G RAM
- Ubuntu 20.04
- Followed the installation doc, and is running on version ba4f2b6b724273a0ae141a6783d28cf5e36ece72

**Demo**
I set up the gradio environment, run with `python app.py --version='GLaMM-FullScope'`, and tried a few examples listed on the page, but the quality is bad (as shown below). No luck if I change `'GLaMM-FullScope'` to other models.

![glamm-issue-1](https://github.com/mbzuai-oryx/groundingLMM/assets/1289856/6379fec9-3dff-426c-9b2a-642f81d73c1b)

![glamm-issue-2](https://github.com/mbzuai-oryx/groundingLMM/assets/1289856/1ba55ce7-2b3e-4022-bbb6-6ea597aabef9)

**Evaluation**
I downloaded COCO train 2014 and refCOCO series, and executed `bash eval/referring_seg/run_evaluation.sh 'MBZUAI/GLaMM-RefSeg' './results_refseg_finetuned'`.

In my initial attempt in numerical evaluation, the code produces an error from the assert [here](https://github.com/mbzuai-oryx/groundingLMM/blob/main/dataset/dataset.py#L220). I looked into it and found `cur_len` is always `total_len + 2`. Without knowledge on how to fix it, I had to comment it out in order to run the script.

Here're the results I have obtained (not finished on every test set, but the performance is obviously bad):

```
[{'model': './results_refseg_finetuned', 'dataset': 'refcoco|val', 'giou': '0.050665364', 'ciou': '0.1175718'}, {'model': './results_refseg_finetuned', 'dataset': 'refcoco|val', 'giou': '0.049141906', 'ciou': '0.11084828'}, {'model': './results_refseg_finetuned', 'dataset': 'refcoco|testA', 'giou': '0.06668462', 'ciou': '0.15553774'}, {'model': './results_refseg_finetuned', 'dataset': 'refcoco|testB', 'giou': '0.034645554', 'ciou': '0.094948955'}, {'model': './results_refseg_finetuned', 'dataset': 'refcoco+|val', 'giou': '0.0536244', 'ciou': '0.11712953'}, {'model': './results_refseg_finetuned', 'dataset': 'refcoco+|testA', 'giou': '0.075434625', 'ciou': '0.15804651'}, {'model': './results_refseg_finetuned', 'dataset': 'refcoco+|testB', 'giou': '0.04812723', 'ciou': '0.110266894'}]
```

Any help or clue to resolve the performance issue is appreciated, thanks!

**Comment by hetong007** [comment]
Thanks @mmaaz60 , having reinstalled the correct version of `transformers` i can reproduce the results with latest CUDA and pytorch.

**Comment by STARRY2001** [comment]
> Thanks @mmaaz60 , having reinstalled the correct version of `transformers` i can reproduce the results with latest CUDA and pytorch.

您好，请问您是如何安装mmcv-full的呢？我用docker管理环境，在pytorch/pytorch:1.13.1-cuda11.6-cudnn8-devel和pytorch/pytorch:2.1.2-cuda12.1-cudnn8-devel建立的容器中，都无法正确安装mmcv-full==1.4.7?
